### DISCUSSION TITLE
Change layout programatically

### DISCUSSION THREAD

**Discussion by bn-sr** [statamic/cms#6007] [Troubleshooting]
Hi,

On v2 I used to be able to swap layouts by setting a config value e.g.

``` Config::set('theming.default_layout', "app"); ```

Is there a way to swap layout on v3? I am trying to swap to an app layout for all pages depending on a query string.

Thanks,

Ben

**Comment by jasonvarga** [comment]
At the moment, no. It's not configurable globally.

You can change it per collection/taxonomy by adding `layout: whatever`.

**Reply by bn-sr** [reply]
Ok, I did manage to do it by setting the layout on the entry collection on the request in middleware.  A global config for it would be nice tho :)
